rfc-163/relocating-logic-from-cdn.md

```diff
@@ -0,0 +1,114 @@
+# Appendix: Proposal for relocating logic from the CDN layer to other parts of the stack
+
+Things that could be moved to WAF:
+
+- Silently ignore certain requests[^drop-requests-1][^drop-requests-2]
+  - This was the outcome of an [incident report](https://docs.google.com/document/d/12DzQsDeu7zUcICy9zVporjprX4qZFIrpOOWtYYRx-nk/edit) - details cannot be provided here, as this is a public repo
+- Serving an HTTP 404 response with a hardcoded template[^autodiscover-template] if the request URL matches `/autodiscover/autodiscover.xml`[^autodiscover-matcher]
+  - Context: https://github.com/alphagov/govuk-cdn-config/pull/86
+- Redirecting `/security.txt` and `/.well-known/security.txt` to `https://vdp.cabinetoffice.gov.uk/.well-known/security.txt`[^redirect-security-txt-1][^redirect-security-txt-2]
```

Comment:
Could this even be done at the application level? I think there's potential we create ourselves another problem of being locked into using AWS WAF. Which would be an issue if we wanted to look at an alt cloud provider - (for example Disaster Recovery)

Reply:
That's a good point actually, I hadn't considered that we'd be moving essentially from one form of vendor lock-in to another.

My concern with e.g. the `security.txt` redirects is that we have the same functionality set up across several different domains (including gov.uk, assets.gov.uk, service.gov.uk and DGU). If we want to move this logic to the application layer then it'll presumably need to be implemented in a different way for each domain, whereas if we move it to the WAF then the implementation is consistent for each.

Reply:
I'd be inclined to do the security.txt redirect in Router, and consider the redirect on service.gov.uk and DGU to be out of scope for this RFC, as they're already not working:

- https://service.gov.uk/security.txt ->